packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx

```diff
@@ -25,11 +26,34 @@ export const BookingFields = ({
   isDynamicGroupBooking: boolean;
 }) => {
   const { t } = useLocale();
-  const { watch, setValue } = useFormContext();
+  const { watch, setValue, getValues } = useFormContext();
   const locationResponse = watch("responses.location");
   const currentView = rescheduleUid ? "reschedule" : "";
   const isInstantMeeting = useBookerStore((state) => state.isInstantMeeting);
 
+  useEffect(() => {
+    const firstLastNameField = fields.find((field) => field.variant === "firstAndLastName");
+
+    if (!firstLastNameField) {
+      return;
+    }
+
+    const variant = firstLastNameField.variant;
+    if (
+      firstLastNameField &&
+      variant &&
+      firstLastNameField.variantsConfig?.variants[variant].fields[1].required
+    ) {
+      const name = getValues("responses.name");
+      if (rescheduleUid && name?.split(" ").length !== 2) {
+        // if the owner of the meeting has changed the lastname to required after the booking was made,
+        // we should allow the booking user to edit the name field
+        setValue("responses.name", "");
```

Comment:
@Amit91848 It's because of how the `react-hook-form` is setup; If we retain the `firstName` and the user goes ahead and just edits the `lastName` since it was empty, the field value would be updated from `somename` to `{lastName: 'lName'}` which then errors out during validation since it expects a `string`.

Making the fields empty and letting the user fill it out **during this edge case scenario** is a cleaner implementation in my opinion. Because of how tightly coupled the validation logic is, it could lead to bugs in other workflows handled correctly. However, if you are familiar with that logic, please feel free to take a stab at it.

Here's a video reference of the issue I've described above:
https://github.com/calcom/cal.com/assets/19223383/5b37c1f0-87db-4073-95d1-9cc7cd3c1bfe